src/common/dav/alloc.h

```diff
@@ -0,0 +1,38 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright 2019-2022, Intel Corporation */
+
+#ifndef COMMON_ALLOC_H
+#define COMMON_ALLOC_H
+
+#include <gurt/common.h>
+
+static inline void *
+Zalloc(size_t sz)
+{
+	void *ptr;
+
+	D_ALLOC(ptr, sz);
+	return ptr;
+}
+
+static inline void *
+Malloc(size_t sz)
+{
+	void *ptr;
+
+	D_ALLOC_NZ(ptr, sz);
+	return ptr;
+}
+
+static inline void *
+Realloc(void *ptr, size_t size)
+{
+	void *new_ptr;
+
+	D_REALLOC_NZ(new_ptr, ptr, size);
+	return new_ptr;
+}
+
+#define	Free(ptr)	D_FREE(ptr)
```

Comment:
One side effect and you can test with D_LOG_MASK=debug DD_MASK=mem is you may get logging for a variable named "ptr" in Malloc but the name will be whatever is passed to the macro here.  That shouldn't cause any issue but may be a little confusing to the log reader.   In such case, it may be better to wrap in an inline function.   I presume the compiler will make it efficient.

What I had in mind with my comment on the prior patch was to wrap the routines, rather than the underlying malloc/free/realloc simiilar to what we do with D_STRNDUP or D_ASPRINTF...the idea being that we log these routines differently.   I'm honestly fine either way though, just clarifying.

Reply:
Even the NLT failures were a result of inconsistent use of D_ALLOC/* macros. There were no real double free getting called. It was just that D_FREE was called on allocations done using posix_mem_align() and strdup(). I have made a new commit where all of the dav routines will now use the DAOS D_ALLOC/FREE/* macros for internal allocation.